### ISSUE TITLE
Compiler doesn't complain on await in an unsafe context.

### ISSUE THREAD

**Issue by EgorBo** [dotnet/roslyn#9889]
**Steps to Reproduce**:

1. Create a partial class Class1 with two files: Class1.A.cs and Class1.B.cs, 
2. in Class1.B.cs mark it with the "unsafe" keyword so the whole class should be "unsafe" now
3. Create an async method MyMethod() in Class1.A.cs and use some await there (like await Task.Delay(1))
4. Compile it - success
5. Move MyMethod() from Class1.A.cs to Class2.B.cs and compile - compilation error.

**Expected Behavior**:
Compilation error in the step 3: "await operator cannot be used in an unsafe context" (like it's doing in the latest Xamarin Studio 6)

**Actual Behavior**:
Compiled successfully.

Small repro project: https://www.dropbox.com/s/d00zc349gcq84nq/UnsafeAwait.zip?dl=0

**Comment by HaloFour** [comment]
@EgorBo 

When you mark only one part of a `partial class` as `unsafe` that doesn't apply to the entire class.  For example, the following is a compiler error:

```cs
public unsafe partial class Class1 {
    public int* field1; // just fine
}

public partial class Class1 {
    // CS0214 Pointers and fixed size buffers may only be used in an unsafe context
    public int* field2;

    public void Foo() {
        // CS0214 Pointers and fixed size buffers may only be used in an unsafe context
        var j = this.field1;
    }

    public unsafe void Bar() {
        var j = this.field1; // just fine
    }
}
```

Because you have the `async` method in the non-`unsafe` part of the class definition that is considered legal.

**Comment by gafter** [comment]
The C# language specification says

> When the `unsafe` modifier is used on a partial type declaration, only that particular part is considered an unsafe context.
